### PR TITLE
Merge in 64-bit branch for game code

### LINKED ISSUE #7: Take on the grinding task of fixing the code for 64bit
if you build this in a 64bit environment it will in fact compile however the game will immediately fail on a malloc() issue or a segfault

Changed 11 months ago by sulucnumoh 

Needs to remove all assumptions that a pointer is a dword.
I tried to reassign dword the better uint32_t from stdint.h but got tons of errors where pointers are cast to void and back again.
But anyway I would go in that direction because it that defines the datatype dword to the datatype the original developer thought of.
But on the other hand this means removal of all parts where a pointer has been thought of the same size as a dword. This is not true anyway.
Should I start with it?

Changed 6 weeks ago by slime73 

priority changed from major to minor
From my understanding, most of the current blocking issues on this stem from the fact that much of the data that the game loads is saved from and loaded into structs which have pointer members. Some of the pointer-members store offset information on disk, which is converted to a real pointer in the same struct when loaded into the game.
Since the size of a pointer is 8 bytes rather than 4 on 64-bit systems, this means that every struct which is filled with info from the disk and has a pointer member will likely have garbled info.
It is not a very big deal to only distribute 32 bit executable versions of the game, s

### PR DESCRIPTION
**Pull request by uberlinuxguy**
I believe that this would close #7

### CODE REVIEW

File: src/libparsec/include/od_odt.h [hunk 1]
```diff
@@ -9,351 +9,459 @@
...
-	ODT_ipol1tex_shad,		// first order interpolated texture mapping (lin)
-	ODT_ipol2tex_shad,		// second order interpolated texture mapping (quad)
-	ODT_persptex_shad,		// perspective correct texmapping w/o interpolation
-	ODT_material_shad,		// shade using material specification
-	ODT_texmat_shad		// composite texture and material specification
-};
+enum ODT_shadingtype_t
+{
+
+	ODT_no_shad,			 // lighting independent ambient color (no shading!)
```
Comment (slime73) [inline src/libparsec/include/od_odt.h:57]: The formatting and code style in a couple of these files got changed a bit but it's not a big deal

File: src/parsec/include/r_gl.h [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 // FIXME: use gles header instead of glew when compiling for OpenGL ES.
 #ifndef GLES
+//
+//  On a linux box use the system path
```
Comment (slime73) [inline src/parsec/include/r_gl.h:13]: Obsolete comment (but nothing that'd stop a merge)

**Comment by CrazySpence** [comment]
Been using it since it came out, good for me

File: src/parsec/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,367 @@
+##
```
Comment (slime73) [inline src/parsec/CMakeLists.txt:1]: Does this work?

**Review by CrazySpence — APPROVED** [review]
Ah this is how you're supposed to do it *clicks*

**Review by slime73 — APPROVED** [review]
I don't think any of my comments necessarily need addressing for the merge

File: src/parsec/CMakeLists.txt (hunk 3 above)
Comment (uberlinuxguy) [reply src/parsec/CMakeLists.txt:1]: I didn't try any of the CMake stuff that @tribad put in, but I didn't see any harm in including it for now...

File: src/libparsec/include/od_odt.h (hunk 1 above)
Comment (uberlinuxguy) [reply src/libparsec/include/od_odt.h:57]: I thought I had fixed it all, but perhaps some of it is still a bit messed up.  I think we can go through and fix it once we agree on some formatting/style stuff.  There may be clean up that is needed elsewhere anyway.

Comment (uberlinuxguy) [reply src/libparsec/include/od_odt.h:57]: It appears to be a difference of tabs vs expanded to space tabs.

**Pull request merged**